sys/sysmon/include/sysmon/sysmon.hpp

```diff
@@ -0,0 +1,87 @@
+/// 
+/// Copyright (C) 2022 COGIP Robotics association <cogip35@gmail.com>
+/// 
+/// This file is subject to the terms and conditions of the GNU Lesser
+/// General Public License v2.1. See the file LICENSE in the top level
+/// directory for more details.
+/// 
+/// 
+/// 
+/// @defgroup    sys_sysmon System monitoring
+/// @ingroup     sys
+/// @brief       System monitoring module
+///              Module used to monitor memory (thread stacks, heap and overall).
+/// 
+/// @{
+/// @file
+/// @brief       Public API for sysmon module
+/// 
+/// @author      Gilles DOFFE <g.doffe@gmail.com>
+/// 
+
+// System includes
+#include <cstddef>
+#include <string>
+
+#pragma once
+
+namespace cogip {
+
+namespace sysmon {
+
+class Memory {
+    public:
+        // Get memory total size in bytes
+        inline std::size_t get_size() { return this->size; };
+        // Get memory free size in bytes
+        inline std::size_t get_used() { return this->used; };
+        // Get memory used size in bytes
+        inline std::size_t get_free() { return this->free; };
+        // Set memory total size in bytes
+        inline void set_size(std::size_t size) { this->size = size; };
+        // Set memory free size in bytes
+        inline void set_used(std::size_t used) { this->used = used; };
+        // Set memory used size in bytes
+        inline void set_free(std::size_t free) { this->free = free; };
+
+    private:
+        // Memory total size in bytes
+        std::size_t size;
+        // Memory free size in bytes
+        std::size_t free;
+        // Memory used size in bytes
+        std::size_t used;
+};
+
+class ThreadStatus: public Memory {
+    public:
+        // Get thread name
+        inline std::string get_name() { return this->name; };
+        // Get thread pid
+        inline uint32_t get_pid() { return this->pid; };
+        // Set thread name
+        inline void set_name(std::string name) { this->name = name; };
+        // Set thread pid
+        inline void set_pid(uint32_t pid) { this->pid = pid; };
+    private:
+        // Thread pid
+        uint32_t pid;
+        // Thread name
+        std::string name;
+};
+
+class Sysmon {
+    public:
+        // Display heap memory status
+        static void display_heap_status(void);
+        // Display each thread status
+        static void display_threads_status(void);
+        // Start system monitoring thread
+        static void sysmon_start(void);
+};
```

Comment:
You do not need a class if you have only static functions in it.

Reply:
done